docs/sdk/SDK_API_EXAMPLES.md

```diff
@@ -0,0 +1,216 @@
+# Mesh Data Plane SDK – Examples
+
+## SMPTE ST2110-22 compressed video sender example
+
+```c
+int main(void)
+{
+    /* Default client configuration */
+    MeshClientConfig client_config = { 0 };
+    
+    /* ST2110-XX configuration */
+    MeshConfig_ST2110 conn_config = {
+        .remote_ip_addr = "192.168.96.2",
+        .remote_port = 9001,
+        .transport = MESH_CONN_TRANSPORT_ST2110_22,
+    };
+
+    /* Video configuration */
+    MeshConfig_Video payload_config = {
+        .width = 1920,
+        .height = 1080,
+        .fps = 60,
+        .pixel_format = MESH_VIDEO_PIXEL_FORMAT_YUV422P10LE,
+    };
+
+    MeshConnection *conn;
+    MeshClient *mc;
+    int err;
+    int i, n;
+
+    /* Create a mesh client */
+    err = mesh_create_client(&mc, &client_config);
+    if (err) {
+        printf("Failed to create mesh client: %s (%d)\n", mesh_err2str(err), err);
+        exit(1);
+    }
+
+    /* Create a mesh connection */
+    err = mesh_create_connection(mc, &conn);
+    if (err) {
+        printf("Failed to create connection: %s (%d)\n", mesh_err2str(err), err);
+        goto exit_delete_client;
+    }
+
+    /* Apply connection configuration */
+    err = mesh_apply_connection_config_st2110(conn, &conn_config);
+    if (err) {
+        printf("Failed to apply SMPTE ST2110 configuration: %s (%d)\n", mesh_err2str(err), err);
+        goto exit_delete_conn;
+    }
+
+    /* Apply video payload configuration */
+    err = mesh_apply_connection_config_video(conn, &video_config);
+    if (err) {
+        printf("Failed to apply video configuration: %s (%d)\n", mesh_err2str(err), err);
+        goto exit_delete_conn;
+    }
+
+    /* Establish a connection for sending data */
+    err = mesh_establish_connection(conn, MESH_CONN_KIND_SENDER);
+    if (err) {
+        printf("Failed to establish connection: %s (%d)\n", mesh_err2str(err), err);
+        goto exit_delete_conn;
+    }
+
+    /* 10 video frames to be sent */
+    n = 10;
+
+    /* Send data loop */
+    for (i = 0; i < n; i++) {
+        MeshBuffer *buf;
```

Comment:
Well, from the resulting assembly code perspective, you won't see any difference if you move the pointer declaration upper. It will even sit in the same memory address on the stack.

There is no `creation` of the pointer in this block. It's just a declaration of a local variable on the stack which disappears when the execution flow exits the block. No memory leakage is possible here.

The only concern I could imagine here is the software developer's habit to declare local variables. From the C standard perspective, it is correct.

But yes, we can move it upper if you'd like.

Reply:
Fixed in eae8b083e0210567994eb379b5e228742451c44b.